client/SMCE_Client.cpp

```diff
@@ -0,0 +1,339 @@
+    // // clang-format on                                                                                                        
+                                                                                                                                 
+    // Create the virtual Arduino board                                                                                          
+    std::cout << "Creating board and compiling sketch" << std::endl;                                                                               
+    smce::Board board(exit_sketch);
+    //Compile and start board
+    compile_and_start(sketch,toolchain,board,arduino_root_dir);                                                                                      
+    std::cout << "Complete" << std::endl;                                                                                                                                                                                                                                          
+    //Create view and uart (serial) channels                                                                                     
+    auto board_view = board.view();                                                                                              
+    auto uart0 = board_view.uart_channels[0];                                                                                    
+                                                                                                                                 
+    //start listener thread for uart                                                                                             
+    std::thread uart_thread{[=] {uart_listener(uart0,file_write,arduino_root_dir);} };                                                                       
+    std::cout << "Messages received on uart from arduino is shown as "
+        << termcolor::red << "red" << termcolor::reset << " text." << std::endl;                                                                                                                             
+    // Print command menu                                                                                                        
+    print_menu();                                                                                                                                                                                                                                              
+    // Main loop, handle the command input                                                                                       
+    bool suspended = false;                                                                                                      
+    while(true){                                                                                                                 
```

Comment:
so I¨m not really going to comment on all below, you can do userinput in a million ways, BUT:
mixing input stuff with core logic is just bad, each action you have in this list of if statements should have an analog in a dedicated class that manages the state of the board, sketch and uart_listener and all the other smce specific things.

Think about how you would even test this program, if you had a class that you can interact with through code it would go much easier.

```cpp
if (input = "do something") {
    object.do_something();
}
```
reads wayy better compared to when the contents of do_something is fully inlined there right?